src/chordsymbol.ts

```diff
@@ -9,51 +9,42 @@
 //
 // See `tests/chordsymbol_tests.ts` for usage examples.
 
-import { Font, FontInfo } from './font';
-import { Glyph } from './glyph';
+import { Element } from './element';
+import { Font } from './font';
 import { Modifier } from './modifier';
 import { ModifierContextState } from './modifiercontext';
 import { Note } from './note';
 import { StemmableNote } from './stemmablenote';
 import { Tables } from './tables';
-import { TextFormatter } from './textformatter';
 import { Category, isStemmableNote } from './typeguard';
-import { log, RuntimeError } from './util';
+import { log } from './util';
 
 // To enable logging for this class. Set `Vex.Flow.ChordSymbol.DEBUG` to `true`.
 // eslint-disable-next-line
 function L(...args: any[]): void {
   if (ChordSymbol.DEBUG) log('Vex.Flow.ChordSymbol', args);
 }
 
-export interface ChordSymbolBlock {
-  text: string;
-  symbolType: SymbolTypes;
+export class ChordSymbolBlock extends Element {
   symbolModifier: SymbolModifiers;
-  xShift: number;
-  yShift: number;
   vAlign: boolean;
-  width: number;
-  glyph?: Glyph;
-}
 
-export interface ChordSymbolGlyphMetrics {
-  leftSideBearing: number;
-  advanceWidth: number;
-  yOffset: number;
-}
+  constructor(text: string, symbolModifier: SymbolModifiers, xShift: number, yShift: number, vAlign: boolean) {
+    super();
+    this.text = text;
+    this.symbolModifier = symbolModifier;
+    this.xShift = xShift;
+    this.yShift = yShift;
+    this.vAlign = vAlign;
+  }
 
-export interface ChordSymbolMetrics {
-  global: {
-    superscriptOffset: number;
-    subscriptOffset: number;
-    kerningOffset: number;
-    lowerKerningText: string[];
-    upperKerningText: string[];
-    spacing: number;
-    superSubRatio: number;
-  };
-  glyphs: Record<string, ChordSymbolGlyphMetrics>;
+  isSuperscript(): boolean {
+    return this.symbolModifier !== undefined && this.symbolModifier === SymbolModifiers.SUPERSCRIPT;
+  }
+
+  isSubscript(): boolean {
+    return this.symbolModifier !== undefined && this.symbolModifier === SymbolModifiers.SUBSCRIPT;
+  }
```

Comment:
In both of these functions, is the undefined check unnecessary?

Can't we just check if the this.symbolModifier === SUPERSCRIPT or SUBSCRIPT? Not sure why we need to check for undefined.

Reply:
you are right, done